### PR TITLE
Normalizer to TTS models, TTS tokenizer updates, AxisKind updates

### PR DESCRIPTION
**Pull request by Oktai15**
* Add normalizer for FastPitch and Mixer-TTS
* Update types for FastPitch and Mixer-TTS via new proposed signature for time
* Fix small bugs and refactor code in Mixer-TTS, FastPitch and TTS tokenizers
* Add German tokenizer
* Add CMUDict

Signed-off-by: Oktai Tatanov <oktai.tatanov@gmail.com>

### CODE REVIEW

File: nemo/collections/tts/models/fastpitch.py [hunk 1]
```diff
@@ -151,6 +153,38 @@ def tb_logger(self):
             self._tb_logger = tb_logger
         return self._tb_logger
 
+    @property
+    def normalizer(self):
+        if self._normalizer is not None:
+            return self._normalizer
+
+        if self.learn_alignment:
+            ds_class_name = self._cfg.train_ds.dataset._target_.split(".")[-1]
```
Comment (blisc) [inline nemo/collections/tts/models/fastpitch.py:162]: Have you checked to see if this works with `from_pretrained`? I think `from_pretrained` deletes `self._cfg.train_ds` and validation

File: nemo/collections/tts/models/fastpitch.py [hunk 2]
```diff
@@ -185,24 +219,28 @@ def parser(self):
             )
         return self._parser
 
-    def parse(self, str_input: str) -> torch.tensor:
+    def parse(self, str_input: str, normalize=True, **kwargs) -> torch.tensor:
```
Comment (blisc) [inline nemo/collections/tts/models/fastpitch.py:222]: Remove **kwargs since it's unused

File: nemo/collections/tts/models/mixer_tts.py [hunk 3]
```diff
@@ -584,7 +604,9 @@ def generate_spectrogram(
         pred_spect = self.infer(tokens, tokens_len, lm_tokens=lm_tokens)
         return pred_spect
 
-    def parse(self, text: str, **kwargs) -> torch.Tensor:
+    def parse(self, text: str, normalize=True, **kwargs) -> torch.Tensor:
```
Comment (blisc) [inline nemo/collections/tts/models/mixer_tts.py:607]: Remove kwargs since its unused

File: examples/tts/conf/mixer-tts.yaml [hunk 4]
```diff
@@ -27,8 +27,9 @@ pitch_loss_scale: 0.1
 durs_loss_scale: 0.1
 mel_loss_scale: 1.0
 
-phoneme_dict_path: "scripts/tts_dataset_files/cmudict-0.7b-030921"
+phoneme_dict_path: "scripts/tts_dataset_files/cmudict-0.7b"
```
Comment (blisc) [inline examples/tts/conf/mixer-tts.yaml:30]: Let's add this back to NeMo. For now, we can use base 07 cmudict. Or whatever we decide in the Wednesday meeting

File: nemo/collections/tts/models/fastpitch.py [hunk 5]
```diff
@@ -185,24 +219,28 @@ def parser(self):
...
 
+        if normalize:
+            str_input = self.normalizer(str_input)
+
         tokens = self.parser(str_input)
 
         x = torch.tensor(tokens).unsqueeze_(0).long().to(self.device)
         return x
 
+    # T1 - text length, T2 - audio length, T3 - spectrogram length
```
Comment (blisc) [inline nemo/collections/tts/models/fastpitch.py:234]: ~~Where in the code is `T1`, T2`, and `T3` defined?~~ Nvm I see you added it in #3090.
I don't like this naming scheme. Maybe we can do something like T_text? Where T represents AxisKind.Time and everything past _ is ignored.
The other problem with the current implementation is that `T1` and `T2` are both treated as AxisKind.Time so you can pass a dimension of `T1` into a type of `T2`. We ought to differentiate between the two.

File: nemo/collections/tts/torch/tts_tokenizers.py [hunk 6]
```diff
@@ -123,16 +84,18 @@ class EnglishCharsTokenizer(BaseTokenizer):
...
         self,
+        chars,
         punct=True,
         apostrophe=True,
         add_blank_at=None,
         pad_with_space=False,
         non_default_punct_list=None,
-        text_preprocessing_func=english_text_preprocessing,
-        word_tokenize_func=english_word_tokenize,
+        text_preprocessing_func=lambda x: x,
```
Comment (blisc) [inline nemo/collections/tts/torch/tts_tokenizers.py:93]: Does it still make sense to have this function if we split normalization and tokenization?

**Review by blisc — CHANGES_REQUESTED** [review]

File: nemo/collections/tts/models/fastpitch.py (hunk 1 above)
Comment (blisc) [reply nemo/collections/tts/models/fastpitch.py:162]: Closed in discussion

File: nemo/collections/tts/models/fastpitch.py (hunk 5 above)
Comment (Oktai15) [reply nemo/collections/tts/models/fastpitch.py]: Fixed it in this PR. I chose chose your idea.

File: examples/tts/conf/mixer-tts.yaml (hunk 4 above)
Comment (Oktai15) [reply examples/tts/conf/mixer-tts.yaml:30]: Fixed.

File: nemo/collections/tts/models/mixer_tts.py (hunk 3 above)
Comment (Oktai15) [reply nemo/collections/tts/models/mixer_tts.py]: Fixed.

File: nemo/collections/tts/models/fastpitch.py (hunk 2 above)
Comment (Oktai15) [reply nemo/collections/tts/models/fastpitch.py]: Fixed.

**Review by blisc — APPROVED** [review]
